### PR TITLE
Functional refactor

### PR DESCRIPTION
**Pull request by eliperelman**
I've been wanting to do a functional architecture for a while in Neutrino, and hopefully this gives us a good base. I'm hoping this leads to make testable units inside the API without so much reliance on state.

### CODE REVIEW

File: packages/neutrino-middleware-clean/index.js [hunk 1]
```diff
@@ -6,5 +6,5 @@ module.exports = (neutrino, options) => {
 
   neutrino.config
     .plugin('clean')
-    .use(CleanPlugin, [paths, { root }]);
+    .use(CleanPlugin, [paths, { root, verbose: false }]);
```
Comment (jaridmargolin) [inline packages/neutrino-middleware-clean/index.js]: 👍 seemed unnecessary to see this on every build

File: packages/neutrino/bin/neutrino [hunk 2]
```diff
@@ -53,26 +50,58 @@ const args = yargs
...
+  const errors = Array.isArray(err) ? err : [err];
 
-  return args.inspect ?
-    inspect(api.getWebpackConfig()) :
-    api[command](args);
-}
+  errors.forEach((err) => {
+    console.error(err.stack || err);
+    err.details && console.error(err.details);
+  });
```
Comment (jaridmargolin) [inline packages/neutrino/bin/neutrino]: Hmmm... I am assuming this was moved from the commands, which I think makes sense, but maybe this should be exposed somewhere, so people consuming the api are capable of easily logging result as well?

File: packages/neutrino/bin/neutrino [hunk 3]
```diff
@@ -53,26 +50,58 @@ const args = yargs
...
-  // Grab all middleware and merge them into a single webpack-chain config instance
-  api.import(middleware);
+  return Object.assign(options, { config, args });
+};
 
-  // Also grab any Neutrino config from package.json and merge it into the config at a higher precedence
-  api.use(() => api.config.merge(config));
+// fail :: Error|Array Error -> IO ()
+const fail = (err) => {
+  const errors = Array.isArray(err) ? err : [err];
```
Comment (jaridmargolin) [inline packages/neutrino/bin/neutrino]: Maybe we could add a `castToArray` method to utils. I believe this is used in multiple places.

Comment (eliperelman) [inline packages/neutrino/bin/neutrino]: Agreed.

File: packages/neutrino/bin/neutrino (hunk 2 above)
Comment (eliperelman) [inline packages/neutrino/bin/neutrino]: Good idea, I'll normalize this to always pass an array of errors.

**Comment by eliperelman** [comment]
Also fixes #129.

**Pull request merged**